test/unit/models/classes/Lists/DataAccess/Repository/ParentPropertyListRepositoryTest.php

```diff
@@ -0,0 +1,168 @@
+    private $range;
+
+    /** @var QueryBuilder|MockObject */
+    private $queryBuilder;
+
+    /** @var RdsValueCollectionRepository|MockObject */
+    private $rdsCollectionRepository;
+
+    /** @var Value|MockObject */
+    private $valueMock;
+
+    /** @var valueCollection|MockObject */
+    private $valueCollection;
+
+    /** @var ComplexSearchService|MockObject */
+    private $complexSearchService;    
+
+    public function setUp(): void
+    {
+        $this->subject = $this->createMock(ParentPropertyListRepository::class);
+        $this->sut = $this->createMock(DependencyRepository::class);
+        $this->range = $this->createMock(core_kernel_classes_Resource::class);
+        $this->queryBuilder = $this->createMock(QueryBuilder::class);
+        $this->rdsCollectionRepository = $this->createMock(RdsValueCollectionRepository::class);
+        $this->valueMock = $this->createMock(Value::class);
+        $this->valueCollection = $this->createMock(ValueCollection::class);
+
+        $platform = $this->createMock(common_persistence_sql_Platform::class);
+        $platform->method('getQueryBuilder')
+            ->willReturn($this->queryBuilder);
+
+        $persistence = $this->createMock(common_persistence_SqlPersistence::class);
+        $persistence->method('getPlatform')
+            ->willReturn($platform);
+
+        $persistenceManager = $this->createMock(PersistenceManager::class);
+        $persistenceManager->method('getPersistenceById')
+            ->willReturn($persistence);
+            $this->sut->setServiceLocator(
+                $this->getServiceLocatorMock(
+                    [
+                        PersistenceManager::SERVICE_ID => $persistenceManager
+                    ]
+                )
+            );
+
+        $this->complexSearchService = $this->createMock(ComplexSearchService::class);
+        $this->subject->setServiceLocator(
+            $this->getServiceLocatorMock(
+                [
+                    ComplexSearchService::SERVICE_ID => $this->complexSearchService
+                ]
+            )
+        );
+    }
+
+    public function testFindAllUris(): void
+    {
+        $collection = new DependencyCollection();
+        $collection->append(new Dependency('uri'));
+
+        $this->range->method('getUri')
+            ->willReturn('list_uri');
+
+        $this->sut
+            ->method('findAll')
+            ->willReturn(
+                $collection
+            );
+
+        $this->valueCollection
+        ->method('getListUris')
+        ->willReturn([]);
+
+        $sampleValue = new Value(null, 'http://sample.com#1', '1');
+        $this->valueMock
+        ->expects(self::any())
+        ->method('setUri')
+        ->with(['uri'])
+        ->willReturn($sampleValue);
+
+        $this->rdsCollectionRepository
+        ->method('findAll')
+        ->with(new ValueCollectionSearchRequest())
+        ->willReturn($this->valueCollection);
```

Comment:
Please, fix indentation

```suggestion
        $this->valueCollection
            ->method('getListUris')
            ->willReturn([]);

        $sampleValue = new Value(null, 'http://sample.com#1', '1');
        $this->valueMock
            ->expects(self::any())
            ->method('setUri')
            ->with(['uri'])
            ->willReturn($sampleValue);

        $this->rdsCollectionRepository
            ->method('findAll')
            ->with(new ValueCollectionSearchRequest())
            ->willReturn($this->valueCollection);
```

Reply:
Done